jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -20,60 +20,135 @@ Starting https://www.docker.com/blog/scaling-docker-to-serve-millions-more-devel
 +
-`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+[source,bash]
+----
+sudo docker run \
+  -d \
+  -p 80:5000 \
+  --restart=always \
+  --name=through-cache \
+  -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" \
+  -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME \
+  -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN \
+  registry
+----
 
-4. Finally, make sure that the TCP port 80 (i.e., HTTP) is open and accessible. For better security, we recommend you open the port only to Nomad clients.
+. Finally, make sure that the TCP port 80 (i.e. HTTP) is open and accessible. For better security, we recommend you to open the port only to Nomad clients, and VMs for `machine` executors and remote docker engines.
 +
 On AWS, for example, you will need to make sure that both Security Groups and the firewall at the OS level are configured correctly to accept connections from Nomad clients over HTTP.
 
-== Configure Nomad clients to use the pull through cache server (run this for _each_ Nomad client)
+=== Set up a secure pull through cache registry (HTTPS proxy for Docker Hub)
+
+Under certain circumstances (especially when you are enabling https://docs.docker.com/develop/develop-images/build_enhancements/[BuildKit] on machine executors and remote docker engines), your pull through cache registry needs to accept connections over HTTPS. This is the instruction to setup a pull through cache that listens for HTTPS connections.
 
-1. Run the command below to *configure the `registry-mirrors` option for the Docker daemon*.
+. *Set up a Docker Hub account* which will be used by the pull through cache registry to access Docker Hub.
 +
-NOTE: Replace `192.0.2.1` with the IP address of your pull through cache server.
+TIP: Setting up a paid account can be a good option to ease the rate limiting further. See also: https://www.docker.com/pricing
+
+. *Set up an independent Linux server* that has Docker installed.
 +
-`sudo bash -c 'cat <<< $(jq ".\"registry-mirrors\" = [\"http://192.0.2.1\"]" /etc/docker/daemon.json) > /etc/docker/daemon.json'`
+We set up this registry as an independent server (i.e. outside of CircleCI boxes) to avoid load on this cache registry affecting other CircleCI Server services.
++
+Note that **the sever needs to be accessible by its hostname**. Namely, you have to configure DNS accordingly so that the hostname (assume `your-mirror.example.com`) resolves to the IP address of the server correctly. Hereby we assume that the URL for the registry to setup is `https://your-mirror.example.com` (be aware that the URL scheme is `http**_s_**`, not `http`). The URL will be needed later to arm Nomad clients and the VM Service.
```

Comment:
```suggestion
NOTE: **the server needs to be accessible by its hostname**. Namely, you must configure DNS accordingly so that the hostname (assume `your-mirror.example.com`) resolves to the IP address of the server correctly. Hereby we assume that the URL for the registry we are setting up is `https://your-mirror.example.com` (be aware that the URL scheme is `http**_s_**`, not `http`). The URL will be needed later to arm Nomad clients and the VM Service.
```